histomicstk/MinimumModel.py

```diff
@@ -0,0 +1,795 @@
+        range e.g. Delta = 0.3 with a uint8 input would translate to 0.3 * 255.
+        Default value = 0.3.
+
+    Notes:
+    ------
+    Objects are assumed to be dark (as nuclei in hematoxylin channel from color
+    deconvolution). Smoothing improves accuracy and computation time by
+    eliminating spurious seed points. Specifying a value for 'Delta' prevents
+    shallow transitions from being included, also reducing computation time and
+    increasing specificity.
+
+    Returns:
+    --------
+    X : array_like
+        A 1D array of horizontal coordinates of contour seed pixels for
+        tracing.
+    Y : array_like
+        A 1D array of the vertical coordinates of seed pixels for tracing.
+    Min : array_like
+        A 1D array of the corresponding minimum values for contour tracing of
+        seed point X, Y.
+    Max : array_like
+        A 1D array of the corresponding maximum values for contour tracing of
+        seed point X, Y.
+
+    See Also:
+    ---------
+    TraceBounds, SeedContours, MinimumModel
+
+    References:
+    -----------
+    .. [1] S. Weinert et al "Detection and Segmentation of Cell Nuclei in
+    Virtual Microscopy Images: A Minimum-Model Approach" in Nature Scientific
+    Reports,vol.2,no.503, doi:10.1038/srep00503, 2012.
+    """
+
+    # initialize outputs
+    X = []
+    Y = []
+    Min = []
+    Max = []
+
+    # for each image row
+    for i in np.arange(I.shape[0]):
+
+        # calculate gradient
+        Gradient = np.hstack((np.nan, I[i, 2:] - I[i, 0:-2], np.nan))
+
+        # identify local maxima and minimia of row 'i' of 'I'
+        Maxima = ((I[i, 1:-1] > I[i, 0:-2]) &
+                  (I[i, 1:-1] > I[i, 2:])).nonzero()[0] + 1
+        Minima = ((I[i, 1:-1] < I[i, 0:-2]) &
+                  (I[i, 1:-1] < I[i, 2:])).nonzero()[0] + 1
+
+        # define max/min neighbor pairs that serve as left-edge contour seeds
+        Merged = np.hstack((Maxima, Minima))
+        Indicator = np.hstack((np.ones((Maxima.size)),
+                               -np.ones((Minima.size))))
+        Indices = np.hstack((np.arange(Maxima.size),
+                             np.arange(Minima.size)))
+        Indicator = Indicator[Merged.argsort()]
+        Indices = Indices[Merged.argsort()]
+        if(Indicator.size > 0):
+            Hits = ((Indicator[0:-1]-Indicator[1:]) == 2).nonzero()[0]
+        Maxima = Maxima[Indices[Hits]]
+        Minima = Minima[Indices[Hits+1]]
+
+        # remove pairs that do not have at least one pixel between them
+        Close = ((Minima - Maxima) < 2).nonzero()
+        Maxima = np.delete(Maxima, Close)
+        Minima = np.delete(Minima, Close)
+
+        # remove pairs that do not have sufficient intensity transitions
+        if(Delta is not None):
+            if np.issubdtype(I.dtype, np.integer):
+                Range = Delta * 255.0
+            elif np.issubdtype(I.dtype, np.float):
+                Range = Delta * 1.0
```

Comment:
`Range = Delta * 1.0` is redundant here.

Maybe it would be nice to normalize the image to [0, 1] range at the beginning of the function before the loop. You can also add the preferred intensity range as an optional parameter. After that you can directly use delta in the range [0, 1].